### PR TITLE
[NPU] update NPU ci tests, test=npu_aarch64

### PR DESCRIPTION
**Pull request by qili93**
### PR types

New features

### PR changes

Others

### Describe

Update paddle ci scripts and add dockerfile for NPU.

aarch64 上的CI结果：https://xly.bce.baidu.com/paddlepaddle/paddle/newipipe/detail/3216137/job/5953440
x86_64 上的CI结果：https://xly.bce.baidu.com/paddlepaddle/paddle/newipipe/detail/3216285/job/5953985

以下单测被disable:
2021-07-21 19:20:12 	134 - c_sync_comm_stream_op_npu_test (SEGFAULT)
2021-07-21 19:20:12 	136 - check_finite_and_unscale_op_npu_test (SEGFAULT)
2021-07-21 19:20:12 	1245 - test_lookup_table_v2_op_npu (Failed)
2021-07-21 19:20:12 	1252 - test_reduce_any_op_npu (Failed)
2021-07-21 19:20:12 	1253 - test_reduce_sum_op_npu (Failed)
2021-07-21 19:20:12 	1270 - test_stack_op_npu (Timeout)
2021-07-22 14:55:36 	  6 - test_bce_loss_npu (Failed)
2021-07-22 14:55:36 	  7 - test_c_identity_npu (Timeout)
2021-07-22 14:55:36 	 43 - test_save_load_npu (Timeout)
2021-07-22 15:13:02 	135 - transpose_op_npu_test (Failed)
2021-07-22 15:13:02 	155 - elementwise_op_npu_test (Failed)

以上单测需要进行修复验证，确认是随机挂还是OP本身的问题

### CODE REVIEW

File: paddle/scripts/paddle_build.sh [hunk 1]
```diff
@@ -1597,6 +1611,42 @@ set -x
...
+                single_card_tests="^$testcase$"
+            else
+                single_card_tests="$single_card_tests|^$testcase$"
+            fi
+        done <<< "$test_cases";
+        card_test "$single_card_tests" 1
+        collect_failed_tests
+set -x
+        ut_endTime_s=`date +%s`
+        echo "XPU testCase Time: $[ $ut_endTime_s - $ut_startTime_s ]s"
```
Comment (zhiqiu) [inline paddle/scripts/paddle_build.sh]: XPU->NPU

File: paddle/scripts/paddle_build.sh [hunk 2]
```diff
@@ -1597,6 +1611,42 @@ set -x
...
+    if [ ${WITH_TESTING:-ON} == "ON" ] ; then
+    cat <<EOF
+    ========================================
+    Running unit npu tests ...
+    ========================================
+EOF
+
+set +x
+        ut_startTime_s=`date +%s`
+        test_cases=$(ctest -N -V | grep "op_npu" )        # cases list which would be run exclusively
```
Comment (zhiqiu) [inline paddle/scripts/paddle_build.sh]: I think we should test all unittests inside `../npu` follder, not only the files ends with `op_npu`.

File: paddle/scripts/paddle_build.sh (hunk 1 above)
Comment (qili93) [reply paddle/scripts/paddle_build.sh]: DONE

File: paddle/scripts/paddle_build.sh (hunk 2 above)
Comment (qili93) [reply paddle/scripts/paddle_build.sh]: DONE

**Review by wanghuancoder — APPROVED** [review]
LGTM

**Pull request merged**